packages/stencil-library/src/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,170 @@
+    MISSING_VALUE: {
+        valid: false,
+        valueMissing: true
+    },
+})
+
+@Component({
+    tag: "njwds-radio-group",
+    styles: `
+        njwds-radio-group .nj-radio-group__validation--message {
+            font-weight: bold;
+            color: #b50909;
+            margin-top: 0.5em;
+        }
+
+        njwds-radio-group .nj-form__required--asterisk {
+            color: #d54309;
+            margin-left: 0.2em;
+        }
+    `
+})
+export class RadioGroup {
+    @Prop() name!: string;
+    @Prop() required: boolean = false;
+    @Prop() tile: boolean = false;
+    @Prop({ reflect: true, mutable: true }) value: string = "";
+    @Prop({ reflect: true, mutable: true }) showValidity: boolean;
+    @Prop() validationMessage: string = "Please select an option.";
+
+    @State() validity: RadioGroupValidityState = validityStates.VALID;
+
+    @Event() njwdsChange: EventEmitter<NjwdsChangeEventDetail>;
+    @Event() njwdsInvalid: EventEmitter<NjwdsInvalidEventDetail>;
+
+    @Element() private hostElement: HTMLStencilElement
+
+    private computeValidity(): RadioGroupValidityState {
+        if (this.required && !this.value) {
+            return validityStates.MISSING_VALUE
+        }
+        return validityStates.VALID
+    }
+
+    private changeErrorPropOnRadios(isError: boolean) {
+        const njwdsRadios = this.hostElement.querySelectorAll("njwds-radio")
+
+        njwdsRadios.forEach((radio: HTMLNjwdsRadioElement) => {
+            radio.error = isError
+        })
+    }
+
+    @Listen('change')
+    changeHandler(event: Event) {
+        if ("value" in event.target && typeof event.target.value === "string") {
+            const newValue = event.target.value
+            this.value = newValue
+            this.njwdsChange.emit({
+                value: newValue
+            })
+        }
+    }
+
+    @Watch('value')
+    watchValueHandler(newValue: string) {
+        const inputs = this.hostElement.querySelectorAll("input")
+        inputs.forEach(input => {
+            const radioValue = input.getAttribute("value")
+            if (radioValue === newValue && !input.checked) {
+                input.checked = true
+            }
+        })
+        this.validity = this.computeValidity()
+        if (this.showValidity) {
+            this.changeErrorPropOnRadios(!this.validity.valid)
+        }
+    }
+
+    @Watch('showValidity')
+    watchShowValidityHandler(newValidity: RadioGroupValidityState) {
+        if (this.showValidity) {
+            this.changeErrorPropOnRadios(!newValidity.valid)
+        }
+    }
+
+    @Method()
+    async getValidity(): Promise<RadioGroupValidityState> {
+        const validityCopy = { ...this.validity }
+        return validityCopy
+    }
+
+    private njwdsInvalidHandler(
+        njwdsInvalidEventEmitter: EventEmitter<NjwdsInvalidEventDetail>,
+    ) {
+        njwdsInvalidEventEmitter.emit({
+            validity: this.validity
+        })
+    }
+
+    componentWillLoad() {
```

Comment:
dust: I kind of want the behavior in `componentWillLoad` and `componentDidLoad` broken out into private functions to more easily separate them out a bit. For instance in `componentWillLoad` we are setting tile attributes and also updating the validity and it feels easier to maintain if they are separated and then run in there.

Reply:
Agreed! The helper method names also add a bit of bonus documentation